config/credentials.default.coffee

```diff
@@ -1,6 +1,33 @@
+generate = () -> Math.random().toString(36).slice(-16)
+
+readOrGenerateSecrets = (file) ->
+  fs = require('fs')
+  secrets = {}
+
+  if fs.existsSync(file)
+    secrets = require(file)
+  else
+    secrets =
+      terraformer: generate()
+      jwt: generate()
+      janitor: generate()
+      vmwatcher: generate()
+      kloud: generate()
+
+    try
+      fs.mkdirSync(require('path').dirname(file))
+    catch err
+      throw err unless err.code is 'EEXIST'
+
+    fs.writeFileSync(file, JSON.stringify(secrets))
+
+  return secrets
+
 module.exports = (options) ->
   kiteHome = "$KONFIG_PROJECTROOT/generated/kite_home/koding"
 
+  secrets = readOrGenerateSecrets("./generated/secrets.json")
```

Comment:
changed require -> readFileSync, now it's relative to CWD which is always top level dir of the repo